### PR TITLE
GODRIVER-2413 Add encryptedFields to Collection.Drop

### PR DESCRIPTION
**Pull request by prestonvasquez**
GODRIVER-2413

## Summary

Add encryptedFields to Collection.Drop

## Background & Motivation

This change is required by DRIVERS-2284 , but was left out of Go Driver 1.x because it could potentially be a backwards breaking change. See [this](https://github.com/mongodb/mongo-go-driver/pull/913#discussion_r868032541) comment on #913  for more context.

### CODE REVIEW

File: mongo/index_view.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ func (iv IndexView) CreateMany(ctx context.Context, models []IndexModel, opts ..
 		Session(sess).WriteConcern(wc).ClusterClock(iv.coll.client.clock).
 		Database(iv.coll.db.name).Collection(iv.coll.name).CommandMonitor(iv.coll.client.monitor).
 		Deployment(iv.coll.client.deployment).ServerSelector(selector).ServerAPI(iv.coll.client.serverAPI).
-		Timeout(iv.coll.client.timeout).MaxTime(option.MaxTime)
+		Timeout(iv.coll.client.timeout).MaxTime(option.MaxTime).Crypt(iv.coll.client.cryptFLE)
```
Comment (kevinAlbs) [inline mongo/index_view.go:279]: Suggest applying a similar change to the `List` and `drop` methods. Those methods also do not pass the `Crypt` methods on the created operations. I expect the operations are not processed for automatic encryption.

Consider filing a separate GODRIVER Bug ticket to track the fix. I expect this bug fix may be desirable to backport into the 1.x driver.

Drivers are expected to process `createIndexes` for automatic encryption. The `createIndexes` command may reference encrypted fields. See https://github.com/mongodb/specifications/pull/1227:

> The "create" and "collMod" commands may contain sensitive data in `validators`:
> 
> ```js
> {
>     "create": "encryptedCollection",
>     "validator": {
>         // "a", "b", and "c" are sensitive.
>         "encryptedField": { "$in": [ "a", "b", "c" ]}
>     }
> }
> ```

Comment (prestonvasquez) [reply mongo/index_view.go:279]: Created GODRIVER-3038 , thanks!

File: mongo/integration/client_side_encryption_test.go [hunk 2]
```diff
@@ -597,3 +597,75 @@ func TestFLE2DocsExample(t *testing.T) {
 		}
 	})
 }
+
+func TestFLE2CreateCollectionWithAutoEncryption(t *testing.T) {
```
Comment (kevinAlbs) [inline mongo/integration/client_side_encryption_test.go:601]:
```suggestion
// `TestFLE2CreateCollectionWithAutoEncryption` is a regression test for a bug fixed in GODRIVER-2413.
// Prior to GODRIVER-2413, the `IndexView.CreateMany` operation was not processed for automatic encryption. This resulted in no "listCollections" command sent.
func TestFLE2CreateCollectionWithAutoEncryption(t *testing.T) {
```

Suggest describing the motivation for this test.

**Review by kevinAlbs — APPROVED** [review]

**Review by blink1073 — APPROVED** [review]
LGTM!

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**